diff -up openmpi-1.10.0/ompi/mca/mtl/ofi/mtl_ofi_component.c.opal openmpi-1.10.0/ompi/mca/mtl/ofi/mtl_ofi_component.c
--- openmpi-1.10.0/ompi/mca/mtl/ofi/mtl_ofi_component.c.opal	2015-08-22 13:19:46.000000000 -0600
+++ openmpi-1.10.0/ompi/mca/mtl/ofi/mtl_ofi_component.c	2015-09-16 13:20:27.749403522 -0600
@@ -13,6 +13,7 @@
  */
 
 #include "mtl_ofi.h"
+#include "opal/util/argv.h"
 
 static int ompi_mtl_ofi_component_open(void);
 static int ompi_mtl_ofi_component_query(mca_base_module_t **module, int *priority);
diff -up openmpi-1.10.0/ompi/tools/ompi_info/ompi_info.c.opal openmpi-1.10.0/ompi/tools/ompi_info/ompi_info.c
--- openmpi-1.10.0/ompi/tools/ompi_info/ompi_info.c.opal	2015-07-16 09:10:59.000000000 -0600
+++ openmpi-1.10.0/ompi/tools/ompi_info/ompi_info.c	2015-09-16 13:24:34.460318311 -0600
@@ -50,6 +50,7 @@
 #endif
 #include "opal/mca/base/base.h"
 #include "opal/runtime/opal_info_support.h"
+#include "opal/util/argv.h"
 #include "opal/util/show_help.h"
 
 #if OMPI_RTE_ORTE
diff -up openmpi-1.10.0/orte/mca/plm/base/plm_base_receive.c.opal openmpi-1.10.0/orte/mca/plm/base/plm_base_receive.c
--- openmpi-1.10.0/orte/mca/plm/base/plm_base_receive.c.opal	2015-07-16 09:10:59.000000000 -0600
+++ openmpi-1.10.0/orte/mca/plm/base/plm_base_receive.c	2015-09-16 13:25:43.628016451 -0600
@@ -37,6 +37,7 @@
 #include "opal/mca/mca.h"
 #include "opal/dss/dss.h"
 #include "opal/threads/threads.h"
+#include "opal/util/argv.h"
 
 #include "orte/constants.h"
 #include "orte/types.h"
