diff -up openmpi-1.8.2rc1/oshmem/tools/wrappers/shmemcc-wrapper-data.txt.in.wrapper openmpi-1.8.2rc1/oshmem/tools/wrappers/shmemcc-wrapper-data.txt.in
--- openmpi-1.8.2rc1/oshmem/tools/wrappers/shmemcc-wrapper-data.txt.in.wrapper	2014-07-11 12:11:56.000000000 -0600
+++ openmpi-1.8.2rc1/oshmem/tools/wrappers/shmemcc-wrapper-data.txt.in	2014-07-17 14:26:03.695430679 -0600
@@ -29,7 +29,7 @@ linker_flags=@OMPI_WRAPPER_EXTRA_LDFLAGS
 # intentionally only link in the MPI libraries (ORTE, OPAL, etc. are
 # pulled in implicitly) because we intend MPI applications to only use
 # the MPI API.
-libs=-loshmem -lmpi @OMPI_WRAPPER_EXTRA_LIBS@
+libs=-loshmem -lmpi
 libs_static=-loshmem -lmpi -lopen-rte -lopen-pal @OMPI_WRAPPER_EXTRA_LIBS@
 dyn_lib_file=libsohmem.@OPAL_DYN_LIB_SUFFIX@
 static_lib_file=liboshmem.a
diff -up openmpi-1.8.2rc1/oshmem/tools/wrappers/shmemfort-wrapper-data.txt.in.wrapper openmpi-1.8.2rc1/oshmem/tools/wrappers/shmemfort-wrapper-data.txt.in
--- openmpi-1.8.2rc1/oshmem/tools/wrappers/shmemfort-wrapper-data.txt.in.wrapper	2014-07-11 12:11:56.000000000 -0600
+++ openmpi-1.8.2rc1/oshmem/tools/wrappers/shmemfort-wrapper-data.txt.in	2014-07-17 14:23:57.148841562 -0600
@@ -28,7 +28,7 @@ linker_flags=@OMPI_WRAPPER_EXTRA_LDFLAGS
 # intentionally only link in the SHMEM and MPI libraries (ORTE, OPAL,
 # etc. are pulled in implicitly) because we intend MPI applications to
 # only use the SHMEM and MPI APIs.
-libs=-loshmem -lmpi_mpifh -lmpi @OMPI_WRAPPER_EXTRA_LIBS@
+libs=-loshmem -lmpi_mpifh -lmpi
 libs_static=-loshmem -lmpi_mpifh -lmpi -lopen-rte -lopen-pal @OMPI_WRAPPER_EXTRA_LIBS@
 dyn_lib_file=liboshmem.@OPAL_DYN_LIB_SUFFIX@
 static_lib_file=liboshmem.a
