--- openmpi-1.2.4/orte/runtime/orte_abort.c.open	2007-10-17 14:47:28.000000000 -0400
+++ openmpi-1.2.4/orte/runtime/orte_abort.c	2007-10-17 14:53:23.000000000 -0400
@@ -88,7 +88,7 @@ int orte_abort(int status, bool report)
             ORTE_ERROR_LOG(ORTE_ERR_OUT_OF_RESOURCE);
             goto CLEANUP;
         }
-        fd = open(abort_file, O_CREAT);
+        fd = open(abort_file, O_WRONLY, O_CREAT);
         if (0 < fd) close(fd);        
     }
     
